storage/index.md

```diff
@@ -70,38 +70,117 @@ PostgreSQL have some multi-modality capability.]
 | PostreSQL     |          | x            |
 | Cassandra     | x        |              |
 | Clickhouse    | x        |              |
-
-
--- write something about how columnar formats are written in some
-   order, and being cognizant of the order that the data, so that you
-   can use the order of the records in the parquet file as an index
-   for ORDER BY and GROUP BY statements.
+| DuckDB        | x        |              |
+
+To avoid needing to read every record to fulfill a query, record-based
+systems use indexes to support more efficient queries. Indexes are
+additional structures with compact representations that contain the
+values of some column (or groups of columns), with pointers to each
+row, that the query can use to limit the number records it must
+inspect. Because indexes are stored in a particular order, queries can
+also use the order of keys in an index to support sorts. While indexes
+are possible for columnar formats in analytical workloads, as is the
+case with Lance, they are less efficient given usage patterns and
+system architectures.
+
+For analytics engines, the encoding and organization of data has a big
+impact on the data. While analytics engines lack indexes in most
+cases, the _order_ of records in a columnar store can be used to avoid
+expensive in-memory sorts, and support otherwise expensive filtering
+operations. While you can run a query over data regardless of speed,
+if you need to improve performance, or want to build a pipeline around
+a query that's slow, you can try to write out or re-materialize the
+out with a different field as the ordering constraint.
+
+In all databases, query optimization is often about finding ways to
+limit the amount of data to consider. In record based systems,
+normalization (splitting logical records across tables,) and indexes
+(materializing compact views) reduces the amount of data to
+consider. For column based systems, saving additional copies of data
+in different forms with different orderings, or that contain a subset
+of fields, can have a huge impact on query performance.
+
+<!-- TODO: connect to "use COPY TO to materialize different views of your data" -->
 
 ### Storage Engines and Encoding
 
-- discuss MVCC and implications for GlareDB, 
-- mention vacuuming requirements
-- discuss encoding efficencies (protobufs and bson)
+GlareDB's data sources can be classified in a few groups:
+
+- structured and semi-structured file formats. There are many
+  different interchange formats, record encoding, and archival formats
```

Comment:
Overall, I really like this section (and think it would make a great blog post for people who are newer to data infrastructure, and don't understand what something like Iceberg is). For the file formats and external database engines paragraphs, could you put the examples up front like you do with the storage engines section? Something like "structures and semi-structured file formats (such as JSON, Parquet, CSV, BSON, and Excel)". I think that will help make these less abstract for readers.